lib/dom/layer.js

```diff
@@ -933,12 +933,20 @@
 
     function AdjustmentLayer(document, group, raw) {
         BaseLayer.call(this, document, group, raw);
+        
+        if (raw.hasOwnProperty("fill")) {
+            this._setFill(raw.fill);
+        }
 
         this._setAdjustment(raw.adjustment);
     }
     util.inherits(AdjustmentLayer, BaseLayer);
 
     Object.defineProperties(AdjustmentLayer.prototype, {
+        "fill": {
+            get: function () { return this._fill; },
+            set: function () { throw new Error("Cannot set fill"); }
```

Comment:
yeah, I wonder how much code was really doing this, it seems a lot of work to really protect all these layer props